lib/auth/result.rb

```diff
@@ -123,6 +123,8 @@ def apply_associated_attributes!
       user.update(associated_group_ids: associated_group_ids)
       AssociatedGroup.where(id: associated_group_ids).update_all("last_used = CURRENT_TIMESTAMP")
     end
+
+    Group.user_trust_level_change!(user.id, user.trust_level) if user && !user.staged
```

Comment:
Why the `!user.staged`? Based on the PR description, I assume this is to refresh the automatic trust-level groups, so perhaps we could add a comment to that effect? `Group.user_trust_level_change!` really sounds like we're changing the user's trust level, even though we're not 😂